differs/package_differs.go

```diff
@@ -71,6 +77,43 @@ func singleVersionDiff(image1, image2 pkgutil.Image, differ SingleVersionPackage
 	}, nil
 }
 
+// singleVersionLayerDiff diffs the packages of each image layer by layer
+func singleVersionLayerDiff(image1, image2 pkgutil.Image, differ SingleVersionPackageLayerAnalyzer) (*util.SingleVersionPackageLayerDiffResult, error) {
+	pack1, err := differ.getPackages(image1)
+	if err != nil {
+		return &util.SingleVersionPackageLayerDiffResult{}, err
+	}
+	pack2, err := differ.getPackages(image2)
+	if err != nil {
+		return &util.SingleVersionPackageLayerDiffResult{}, err
+	}
+	var pkgDiffs []util.PackageDiff
+
+	// Go through each layer for image1
+	for i := range pack1 {
+		if i >= len(pack2) {
+			// Skip diff when there is no layer to compare with in image2
+			continue
+		}
+
+		pkgDiff := util.GetMapDiff(pack1[i], pack2[i])
+		pkgDiffs = append(pkgDiffs, pkgDiff)
+	}
+
+	if len(image1.Layers) != len(image2.Layers) {
+		logrus.Infof("%s and %s have different number of layers, please consider using container-diff analyze to view the contents of each image in each layer", image1.Source, image2.Source)
```

Comment:
So what does the output look like for this case? We should maybe consider either a) not running the diff at all, or b) trying to do something smart to figure out which layers "match up" between the images. I could also be convinced to just run it as normal, but I'm not sure how useful the output will be.

Reply:
I agree that this diff is likely to be useless unless you know in advance the images are somehow comparable and they are build using an equivalent procedure (same number of layers, layers organized in the same way, etc.). This is pretty unlikely to happen, agree. I included it for completeness and because this is the approach that is followed in the `layer` diff. 

I understand the concerns, IMHO this is the same issue exposed in #239. As I already suggested in the mentioned issue, I believe that probably rather than trying find out some smart algorithm to choose the layers to diff it could be easier to just let user choose which layers to diff and let user determine which diff is actually meaningful or meaningless.

I am also opened to just not perform this diff at all as you suggest in a).

Reply:
I think we should just go ahead and not run it here, the output will likely just be meaningless so I'd rather not add functionality that can be potentially confusing. I do like the idea of letting the user choose the specific layers to diff as an alternative though, I'll comment on the other issue.

Reply:
Ok, I fully agree. So for now I'll modify it to not run the diff here and in case #239 we would be still on time to discuss again if it makes sense to run this diff or not.